tests/semantic_analyser.cpp

```diff
@@ -506,6 +507,23 @@
   test("kprobe:f { @x = 1; delete(@x) ? 0 : 1; }", 10);
 }
 
+TEST(semantic_analyser, call_delete_wildcard)
+{
+  test("kprobe:f { @x[0, 1] = 1; delete(@x[*, 1]); }", 0);
+  test("kprobe:f { @x[0, 1] = 1; $y = 1; delete(@x[* ,$y]); }", 0);
+  test("kprobe:f { @x[0, 1] = 1; @y = delete(@x[10, *]); }", 1);
+  test("kprobe:f { @x[0, 1] = 1; @[delete(@x[10, *])] = 1; }", 1);
+  test("kprobe:f { @x[0, 1] = 1; if(delete(@x[10, *])) { 123 } }", 10);
+  test("kprobe:f { @x[0, 1] = 1; delete(@x[10, *]) ? 0 : 1; }", 10);
```

Comment:
Add a negative test case for lone wildcard (ie `delete(@x[*])`) - should error and say "use clear()` isntead ?